internal/api/v1/errors.go

```diff
@@ -103,6 +103,14 @@ func NewNotFoundError(msg string, details ...string) APIError {
 	return NewAPIError(msg, strings.Join(details, ", "), http.StatusNotFound)
 }
 
+// UserNotFound constructs an API error for when the user name is not found in the header
+func UserNotFound() APIError {
+	return NewAPIError(
+		"User not found in the request header",
+		"",
+		http.StatusNotFound)
```

Comment:
In my opinion this error is not a 404 (which is about the addressed resource). I believe what we have here is a `http.BadRequest` instead, because the missing required header makes the request malformed.

Reply:
Done.